From 3b023e571cf2c13cb13c7e299080dca507a38f38 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 5 Aug 2015 18:33:49 +0800
Subject: [PATCH 0223/5242] MLK-11326 dma: mxs-dma: remove the runtime pm
 dummy code

commit  34dc4b6c3e6d756c1d347da0669baa8b26b4a6dc from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, the dma engine driver don't support runtime pm,
and it is not necessary to support the feature since it support
slave sg and cyclic mode, and clock enable/disable during dma
chans allocate and release.

The patch remove the runtime pm dummy code.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: 2c8f8e3e6a21184e6cf8b8e5ba3ec8e76794c951)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/mxs-dma.c |   35 -----------------------------------
 1 file changed, 35 deletions(-)

diff --git a/drivers/dma/mxs-dma.c b/drivers/dma/mxs-dma.c
index d5e7c2a..0704ded 100644
--- a/drivers/dma/mxs-dma.c
+++ b/drivers/dma/mxs-dma.c
@@ -24,7 +24,6 @@
 #include <linux/of_device.h>
 #include <linux/of_dma.h>
 #include <linux/list.h>
-#include <linux/pm_runtime.h>
 #include <asm/irq.h>
 
 #include "dmaengine.h"
@@ -905,39 +904,6 @@ static int __init mxs_dma_probe(struct platform_device *pdev)
 	return 0;
 }
 
-static int mxs_dma_runtime_suspend(struct device *dev)
-{
-	struct mxs_dma_engine *mxs_dma = dev_get_drvdata(dev);
-
-	if (mxs_dma->dev_id == IMX7D_DMA)
-		clk_disable(mxs_dma->clk_io);
-
-	clk_disable(mxs_dma->clk);
-	return 0;
-}
-
-static int mxs_dma_runtime_resume(struct device *dev)
-{
-	struct mxs_dma_engine *mxs_dma = dev_get_drvdata(dev);
-	int ret;
-
-	ret = clk_enable(mxs_dma->clk);
-	if (ret < 0)
-		goto err_out;
-
-	if (mxs_dma->dev_id == IMX7D_DMA) {
-		ret = clk_enable(mxs_dma->clk_io);
-		if (ret < 0)
-			goto err_out;
-	}
-
-	return 0;
-
-err_out:
-	dev_err(dev, "clk_enable failed: %d\n", ret);
-	return ret;
-}
-
 static int mxs_dma_pm_suspend(struct device *dev)
 {
 	/*
@@ -959,7 +925,6 @@ static int mxs_dma_pm_resume(struct device *dev)
 }
 
 static const struct dev_pm_ops mxs_dma_pm_ops = {
-	SET_RUNTIME_PM_OPS(mxs_dma_runtime_suspend, mxs_dma_runtime_resume, NULL)
 	SET_SYSTEM_SLEEP_PM_OPS(mxs_dma_pm_suspend, mxs_dma_pm_resume)
 };
 
-- 
1.7.9.5

